cmd/nitro/init.go

```diff
@@ -406,6 +406,26 @@ func isLeveldbNotExistError(err error) bool {
 	return os.IsNotExist(err)
 }
 
+func databaseIsEmpty(db ethdb.Database) bool {
+	it := db.NewIterator(nil, nil)
+	defer it.Release()
+	return !it.Next()
+}
+
+// if db is not empty, validates if wasm database schema version matches current version
+// otherwise persists current version
+func validateOrWriteWasmStoreSchemaVersion(db ethdb.Database) error {
+	if !databaseIsEmpty(db) {
+		version := rawdb.ReadWasmSchemaVersion(db)
+		if len(version) != 1 || version[0] != rawdb.WasmSchemaVersion {
+			return fmt.Errorf("Wasm database schema version doesn't match current version, current: %v, read from wasm database: %v", rawdb.WasmSchemaVersion, version)
```

Comment:
this version is sort of backwards-compatible with previous one (i.e. it will seem empty). If the version key doesn't exist I think it's o.k. to just write version 1 in that case.
We should only err here if version is more than what we support.

Reply:
I made the change to error only for newer versions, but I have second thoughts.

Wouldn't it be worth to error when version 0 is detected so the user would need to remove the wasm database? Or will we handle that when handling pruning of the wasm database?

Reply:
This should be in a version released before arb1 mainnet, so we should almost never actually need to upgrade to version 1 "in real life".
If that wasn't the case you were probably right, but as it is I prefer to err on the side of easier to operate.